deployment/stacks/ui.ts

```diff
@@ -0,0 +1,239 @@
+import { Construct } from "constructs";
+import {
+  aws_cloudfront as cloudfront,
+  aws_cloudfront_origins as cloudfrontOrigins,
+  aws_iam as iam,
+  aws_s3 as s3,
+  Duration,
+  RemovalPolicy,
+  aws_certificatemanager as acm,
+} from "aws-cdk-lib";
+import { WafConstruct } from "../constructs/waf";
+import { addIamPropertiesToBucketAutoDeleteRole } from "../utils/s3";
+import { IManagedPolicy } from "aws-cdk-lib/aws-iam";
+import { isLocalStack } from "../local/util";
+
+interface CreateUiComponentsProps {
+  scope: Construct;
+  stage: string;
+  project: string;
+  isDev: boolean;
+  iamPermissionsBoundary: IManagedPolicy;
+  iamPath: string;
+  cloudfrontCertificateArn?: string;
+  cloudfrontDomainName?: string;
+  vpnIpSetArn?: string;
+  vpnIpv6SetArn?: string;
+}
+
+export function createUiComponents(props: CreateUiComponentsProps) {
+  const {
+    scope,
+    stage,
+    project,
+    isDev,
+    iamPermissionsBoundary,
+    iamPath,
+    cloudfrontCertificateArn,
+    cloudfrontDomainName,
+    /*
+     * vpnIpSetArn,
+     * vpnIpv6SetArn,
+     */
+  } = props;
+
+  // S3 Bucket for UI hosting
+  const uiBucket = new s3.Bucket(scope, "uiBucket", {
+    encryption: s3.BucketEncryption.S3_MANAGED,
+    removalPolicy: RemovalPolicy.DESTROY,
+    autoDeleteObjects: true,
+    enforceSSL: true,
+  });
+
+  const logBucket = new s3.Bucket(scope, "CloudfrontLogBucket", {
+    encryption: s3.BucketEncryption.S3_MANAGED,
+    publicReadAccess: false,
+    blockPublicAccess: s3.BlockPublicAccess.BLOCK_ALL,
+    objectOwnership: s3.ObjectOwnership.BUCKET_OWNER_PREFERRED,
+    removalPolicy: isDev ? RemovalPolicy.DESTROY : RemovalPolicy.RETAIN,
+    autoDeleteObjects: isDev,
+    enforceSSL: true,
+  });
+
+  // Add bucket policy to allow CloudFront to write logs
+  logBucket.addToResourcePolicy(
+    new iam.PolicyStatement({
+      effect: iam.Effect.ALLOW,
+      principals: [new iam.ServicePrincipal("cloudfront.amazonaws.com")],
+      actions: ["s3:PutObject"],
+      resources: [`${logBucket.bucketArn}/*`],
+    })
+  );
+
+  const securityHeadersPolicy = new cloudfront.ResponseHeadersPolicy(
```

Comment:
In SEDS we made this retain if !isDev but I don't think we should ever retain this since it doesn't contain any app state just the config as structured below which should be easily recreateable.

Reply:
I suspect the logs are required to be retained.

Reply:
I haven't seen any logs associated with this. It is just a policy I think. Let me know if I'm wrong and there is a log group associated with this that I'm missing.